src/EntityGraphQL/Compiler/EntityQuery/EntityQueryNodeVisitor.cs

```diff
@@ -149,12 +148,17 @@ public override Expression VisitIdentity(EntityQLParser.IdentityContext context)
                         return Expression.Default(enumType.TypeDotnet);
                     }
 
-
                     throw new EntityGraphQLCompilerException($"Field {field} not found on type {schemaType.Name}");
                 }
 
                 return enumOrConstantValue;
             }
+
+            if(schemaType.IsEnum)
+            {
+                return Expression.Constant(Enum.Parse(schemaType.TypeDotnet, context.GetText()));                
+            }
+
```

Comment:
The `ConstantVisitor` is trying to handle this.

https://github.com/EntityGraphQL/EntityGraphQL/blob/master/src/EntityGraphQL/Compiler/ConstantVisitor.cs#L74

I wonder why it is returning `null`?

Reply:
It skips the block the second time through due to the schema.hasfield check.

I'd get the constant visitor to do this but it doesn't have the schema type passed to it, so seemed easier inline.

Only covers the case of a fully qualified enum.